src/commands/edge_app.rs

```diff
@@ -185,6 +185,37 @@ impl EdgeAppCommand {
         Ok(())
     }
 
+    pub fn promote(&self, manifest: &EdgeAppManifest, version: &Option<i32>) -> Result<i32, CommandError> {
```

Comment:
just pass manifest - manifest has revision. If user sets version in the cli, modify manifest struct

Reply:
This version could be different than in the manifest - for example downgrading to some previous version.